lib/net.js

```diff
@@ -125,6 +125,10 @@ function normalizeArgs(args) {
   const arg0 = args[0];
   var options = {};
   if (typeof arg0 === 'object' && arg0 !== null) {
+    // Handle the case where the arguments were already normalized.
+    if (Array.isArray(arg0))
```

Comment:
Actually, in my experiences `Array.isArray` is faster because of the difficult-to-optimize nature of the `instanceof` operator under ES2015+…